service/src/main/kotlin/fi/espoo/evaka/nekku/NekkuService.kt

```diff
@@ -0,0 +1,100 @@
+// SPDX-FileCopyrightText: 2017-2025 City of Espoo
+//
+// SPDX-License-Identifier: LGPL-2.1-or-later
+package fi.espoo.evaka.nekku
+
+import com.fasterxml.jackson.databind.json.JsonMapper
+import com.fasterxml.jackson.module.kotlin.readValue
+import fi.espoo.evaka.NekkuEnv
+import fi.espoo.evaka.shared.async.AsyncJob
+import fi.espoo.evaka.shared.async.AsyncJobRunner
+import fi.espoo.evaka.shared.async.AsyncJobType
+import fi.espoo.evaka.shared.async.removeUnclaimedJobs
+import fi.espoo.evaka.shared.db.Database
+import fi.espoo.evaka.shared.domain.EvakaClock
+import io.github.oshai.kotlinlogging.KotlinLogging
+import java.io.IOException
+import okhttp3.OkHttpClient
+import okhttp3.Request
+import okhttp3.Response
+import org.springframework.stereotype.Service
+
+private val logger = KotlinLogging.logger {}
+
+val loggerWarner: (String) -> Unit = { s -> logger.warn { s } }
+
+@Service
+class NekkuService(
+    env: NekkuEnv?,
+    jsonMapper: JsonMapper,
+    private val asyncJobRunner: AsyncJobRunner<AsyncJob>,
+) {
+    private val client = env?.let { NekkuHttpClient(it, jsonMapper) }
+
+    init {
+        asyncJobRunner.registerHandler(::syncNekkuCustomers)
+    }
+
+    fun syncNekkuCustomers(
+        db: Database.Connection,
+        clock: EvakaClock,
+        job: AsyncJob.SyncNekkuCustomers,
+    ) {
+        if (client == null) error("Cannot sync Nekku customers: NekkuEnv is not configured")
+        fetchAndUpdateNekkuCustomers(client, db, loggerWarner)
+    }
+
+    fun planNekkuCustomersSync(db: Database.Connection, clock: EvakaClock) {
+        db.transaction { tx ->
+            tx.removeUnclaimedJobs(setOf(AsyncJobType(AsyncJob.SyncNekkuCustomers::class)))
+            asyncJobRunner.plan(
+                tx,
+                listOf(AsyncJob.SyncNekkuCustomers()),
+                runAt = clock.now(),
+                retryCount = 1,
+            )
+        }
+    }
+}
+
+interface NekkuClient {
+
+    fun getCustomers(): List<NekkuCustomer>
+}
+
+class NekkuHttpClient(private val env: NekkuEnv, private val jsonMapper: JsonMapper) : NekkuClient {
+    val client = OkHttpClient()
+
+    override fun getCustomers(): List<NekkuCustomer> = request(env, "customers")
```

Comment:
OkHttp:n kanssa ei tarvita erillistä `request` utility funktiota, vaan sillä on siistimpää tehdä request suoraan `getCustomers` funktiossa:
```suggestion
    override fun getCustomers(): List<NekkuCustomer> {
        val fullUrl = env.url.resolve("customers").toString()

        val request =
        //... loput request funktiosta
```